tests/conftest.py

```diff
@@ -33,12 +34,20 @@
 from data_safe_haven.logging import init_logging
 
 
+def pytest_configure():
+    """Define constants for use across multiple tests"""
+    pytest.guid_admin = "00edec65-b071-4d26-8779-a9fe791c6e14"
+    pytest.guid_entra = "48b2425b-5f2c-4cbd-9458-0441daa8994c"
+    pytest.guid_subscription = "35ebced1-4e7a-4c1f-b634-c0886937085d"
+    pytest.guid_tenant = "d5c5c439-1115-4cb6-ab50-b8e547b6c8dd"
```

Comment:
I'd not seen `pytest_configure` before :rocket:.

It might not be a problem, but adding attributes to pytest feels a little odd.
Is there a reason to not use a fixture or fixtures?

Reply:
This line `.replace("guid_subscription", pytest.guid_subscription)` doesn't work with a fixture. The `replace` function complains that `guid_subscription` is a function, but `guid_subscription()` isn't treated correctly by `pytest`.

This is a pattern that I've seen recommended in a few places e.g. here (https://pytest-with-eric.com/hooks/pytest-configure/#Share-Global-Variables-with-pytest-configure).

I agree - it would be nicer to simply use test-suite-level variables, but these don't seem to work as expected.

Reply:
:laugh: I was looking at the exact same site. I almost put in my comment that adding the attributes to `config` rather than `pytest` feels _a bit_ more safe.